endpoints/cookie_sync.go

```diff
@@ -108,7 +108,19 @@ func (deps *cookieSyncDeps) Endpoint(w http.ResponseWriter, r *http.Request, _ h
 	}
 
 	parsedReq.filterExistingSyncs(deps.syncers, userSyncCookie)
+	adapterSyncs := make(map[openrtb_ext.BidderName]bool)
+	for _, b := range parsedReq.Bidders {
+		// assume all bidders will be GDPR blocked
+		adapterSyncs[openrtb_ext.BidderName(b)] = true
+	}
 	parsedReq.filterForGDPR(deps.syncPermissions)
+	for _, b := range parsedReq.Bidders {
+		// surviving bidders are not GDPR blocked
+		adapterSyncs[openrtb_ext.BidderName(b)] = false
+	}
+	for b, g := range adapterSyncs {
+		deps.metrics.RecordAdapterCookieSync(b, g)
+	}
 	parsedReq.filterToLimit()
```

Comment:
This logic might make some the metrics misleading...  cause (as an outsider) I would probably expect the `/cookie_sync allowed by GDPR` numbers to match up with the `/setuid attempts` numbers.

Or at least, it seems _useful_ if the numbers lined up, cause we could use them to diagnose bidders with broken sync endpoints.

Not sure how you want to address these, or if it doesn't matter much to you... but maybe add another `true/false` label for `limited_by_count`?

I can also understand the argument for limiting the number of labels/metrics for performance,  though, so... your call.

Reply:
Well, there is a problem with limited by count. The client could implement its own "limit syncs" function after getting the syncs, especially if they were not aware of the functionality within the endpoint. Or may slowly fire off the syncs one at a time and may not get them all off before a page load breaks the chain and triggers a new `/cookie-sync` call. So I don't expect it will be possible for the counts to line up even accounting for `limited_by_count`.

Reply:
ah... yeah, good points.